### PR TITLE
Fix bug in Fetch earliest offsets

### PR DESCRIPTION
**Pull request by bkim54**
when offset isnt ready need to call proper method in setTimeout.

### CODE REVIEW

**Comment by bkim54** [comment]
@hyperlink requested changes made but your tests fail due to the changes
image

**Comment by hyperlink** [comment]
Right, those tests should fail because they rely on the ready var being changed. Can you update the test to emit `ready` from client instead of changing the value? Emitting should also implicitly set the ready to true.

File: lib/offset.js [hunk 1]
```diff
@@ -23,9 +23,7 @@ util.inherits(Offset, events.EventEmitter);
 
 Offset.prototype.fetch = function (payloads, cb) {
   if (!this.ready) {
-    setTimeout(function () {
-      this.fetch(payloads, cb);
-    }.bind(this), 100);
+    this.once('ready', () => this.fetch(payloads, cb).bind(this));
```
Comment (hyperlink) [inline lib/offset.js]: We don't need a bind here since we're using fat arrow.

File: lib/offset.js [hunk 2]
```diff
@@ -43,19 +41,15 @@ Offset.prototype.buildPayloads = function (payloads) {
 
 Offset.prototype.commit = function (groupId, payloads, cb) {
   if (!this.ready) {
-    setTimeout(function () {
-      this.commit(groupId, payloads, cb);
-    }.bind(this), 100);
+    this.once('ready', () => this.commit(groupId, payloads, cb).bind(this));
```
Comment (hyperlink) [inline lib/offset.js]: ditto here.

File: lib/offset.js [hunk 3]
```diff
@@ -43,19 +41,15 @@ Offset.prototype.buildPayloads = function (payloads) {
...
   }
   this.client.sendOffsetCommitRequest(groupId, this.buildPayloads(payloads), cb);
 };
 
 Offset.prototype.fetchCommits = function (groupId, payloads, cb) {
   if (!this.ready) {
-    setTimeout(function () {
-      this.fetchCommits(groupId, payloads, cb);
-    }.bind(this), 100);
+    this.once('ready', () => this.fetchCommits(groupId, payloads, cb).bind(this));
```
Comment (hyperlink) [inline lib/offset.js]: and here.

**Comment by bkim54** [comment]
@hyperlink, good to go?

**Comment by hyperlink** [comment]
Published as 1.3.3.

**Pull request merged**